File: java/src/com/vo/Counter.java
package com.vo;

import java.time.ZonedDateTime;

public class Counter {
    private String ip;
    private String name;
    private Integer totalIn = 0;
    private Integer totalOut = 0;
    private String friendlyName = "";
    private ZonedDateTime lastUpdate;
    private String serial = "";

    public Counter(String ip, String name) {
        this.ip = ip;
        this.name = name;
    }

    public void update() {
        try {
            Response r = new FakeHttp().get("http://" + ip + "/local/people-counter/.api?live-sum.json");
            totalIn = r.in();
            totalOut = r.out();
            friendlyName = r.name();
            lastUpdate = r.timestamp();
            serial = r.serial();
        } catch (Exception e) {
            friendlyName = "error getting count from " + name;
        }
    }

    public String getIp() { return ip; }
    public String getName() { return name; }
    public Integer getTotalIn() { return totalIn; }
    public Integer getTotalOut() { return totalOut; }
    public String getFriendlyName() { return friendlyName; }
    public ZonedDateTime getLastUpdate() { return lastUpdate; }
    public String getSerial() { return serial; }
}

File: csharp/people-counter/Domain/Counter.cs
using System;
using System.Data;

namespace Domain.Tests
{
    public class Counter
    {
        private readonly HttpClient _httpClient;
        private readonly string _name;
        private readonly string _ip;
        public int TotalIn;
        public int TotalOut;
        private string _friendlyName;
        private DateTime _lastUpdate;
        private string _serial;

        public Counter(HttpClient httpClient, string name, string ip)
        {
            _httpClient = httpClient;
            _name = name;
            _ip = ip;
            this.Update();
        }

        public void Update()
        {
            try
            {
                var data = this._httpClient.fetch("http://${ip}/people-counter/api/live.json");
                this.TotalIn = data.InAmount;
                this.TotalOut = data.OutAmount;
                this._friendlyName = data.Name;
                this._lastUpdate = data.Timestamp;
                this._serial = data.Serial;
            } catch (Exception)
            {
                this._friendlyName = "error connectiong to ${name}";
            }
        }
    }
}